flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -2260,6 +2263,121 @@ public void testSavepointScheduledInUnalignedMode() throws Exception {
+					AcknowledgeCheckpoint acknowledgeCheckpoint1 = new AcknowledgeCheckpoint(
+						jid, attemptID1, checkpointId, new CheckpointMetrics(), taskOperatorSubtaskStates1);
+					AcknowledgeCheckpoint acknowledgeCheckpoint2 = new AcknowledgeCheckpoint(
+						jid, attemptID2, checkpointId, new CheckpointMetrics(), taskOperatorSubtaskStates2);
+					coord.receiveAcknowledgeMessage(acknowledgeCheckpoint1, TASK_MANAGER_LOCATION_INFO);
+					coord.receiveAcknowledgeMessage(acknowledgeCheckpoint2, TASK_MANAGER_LOCATION_INFO);
+					return null;
+				}
+
+				@Override
+				public void restoreCheckpoint(long checkpointId, Integer checkpointData) throws Exception {
+
+				}
+
+				@Override
+				public SimpleVersionedSerializer<Integer> createCheckpointDataSerializer() {
+					return new SimpleVersionedSerializer<Integer>() {
+						@Override
+						public int getVersion() {
+							return 0;
+						}
+
+						@Override
+						public byte[] serialize(Integer obj) throws IOException {
+							return new byte[0];
+						}
+
+						@Override
+						public Integer deserialize(int version, byte[] serialized) throws IOException {
+							return null;
+						}
+					};
+				}
+			});
+
+			assertEquals(0, coord.getNumberOfPendingCheckpoints());
+			assertEquals(0, coord.getNumberOfRetainedSuccessfulCheckpoints());
+			assertEquals(0, manuallyTriggeredScheduledExecutor.getScheduledTasks().size());
+
+			// trigger the first checkpoint. this should succeed
+			final CompletableFuture<CompletedCheckpoint> checkpointFuture = coord.triggerCheckpoint(false);
+			manuallyTriggeredScheduledExecutor.triggerAll();
+			assertFalse(checkpointFuture.isCompletedExceptionally());
+
+			// the now we should have a completed checkpoint
+			assertEquals(1, coord.getNumberOfRetainedSuccessfulCheckpoints());
+			assertEquals(0, coord.getNumberOfPendingCheckpoints());
+
+			// the canceler should be removed now
+			assertEquals(0, manuallyTriggeredScheduledExecutor.getScheduledTasks().size());
+
+			// validate that the subtasks states have registered their shared states.
+			verify(subtaskState1, times(1)).registerSharedStates(any(SharedStateRegistry.class));
+			verify(subtaskState2, times(1)).registerSharedStates(any(SharedStateRegistry.class));
+
+			// validate that the relevant tasks got a confirmation message
+			long checkpointId = checkpointIdRef.get();
+			verify(vertex1.getCurrentExecutionAttempt(),
+				times(1)).triggerCheckpoint(eq(checkpointId), any(Long.class), any(CheckpointOptions.class));
+			verify(vertex2.getCurrentExecutionAttempt(),
+				times(1)).triggerCheckpoint(eq(checkpointId), any(Long.class), any(CheckpointOptions.class));
+
+			CompletedCheckpoint success = coord.getSuccessfulCheckpoints().get(0);
+			assertEquals(jid, success.getJobId());
+			assertEquals(2, success.getOperatorStates().size());
+
+			coord.shutdown(JobStatus.FINISHED);
+		}
+		catch (Throwable e) {
+			e.printStackTrace();
+			fail(e.getMessage());
+		}
```

Comment:
This way of handling and catching of an exception is a left over of some workaround of an old bug (JUnit? Travis? I have no idea). Nowadays there is no need for this, so you could remove this try/catch block - it would also improve error message, as you wouldn't be hiding the original stack trace.